sda_orchestrator/utils/id_ops.py

```diff
@@ -1,31 +1,185 @@
 """Fetching IDs for files and datasets."""
 
 from pathlib import Path
+from typing import Dict, Union
 from uuid import uuid4
+from os import environ
+from datetime import date
+import shortuuid
 
+from .logger import LOG
+from ..config import CONFIG_INFO
 
-def generate_dataset_id(user: str, inbox_path: str) -> str:
+from httpx import Headers, AsyncClient, Response, DecodingError
+
+
+def generate_dataset_id(user: str, inbox_path: str, ns: Union[str, None] = None) -> str:
     """Map accession ID to dataset.
 
     Generate dataset id based on folder or user.
+    We keep the email domain as users might have same name on different domains
+    and that could indicate different datasets.
     """
     file_path = Path(inbox_path)
     file_path_parts = file_path.parts
     dataset = ""
+    ns = ns if ns else "urn:neic:"
+
+    # add trailing slash if it does not exist
+    if ns.startswith(("http://", "https://")):
+        if ns[len(ns) - 1] != "/":
+            ns = ns + "/"
```

Comment:
```suggestion
        ns = ns.rstrip("/") + "/"
```
We can omit an `if`-block and a math operation by always stripping and appending the slash.